tests/net/trickle/testcase.yaml

```diff
@@ -4,4 +4,4 @@ common:
 tests:
   net.trickle:
     min_ram: 12
-    tags: net trickle
+    tags: net trickle emu_time
```

Comment:
There was some issues with Trickle test earlier (like months ago) but I though I had changed the tests slightly to "fix" the issues and I have not seen (at least in networking PRs) that the Trickle tests are failing. Do you have a link to sanitycheck log about the failure case?

Reply:
@jukkar are you saying these tests do not require a reliable clock or not any more?

Reply:
> are you saying these tests do not require a reliable clock or not any more?

I was not saying that (at least not yet :), I was just wondering if this is needed because personally I have not seen issues with Trickle tests lately.

Reply:
This emu_time tag is purely about marking tests that require a reliable clock. Have you observed clock issues in general? For instance have you seen any other test in this list failing sometimes with timing related errors?

- If not then your system somehow provides tests a reliable clock. See references in the commit message.
- If you do observe on your system random timing failures spread across various tests then I may have mistaken a past failure of this trickle test for an emu_time issue.

Reply:
I removed this emu_time tag on `tests/net/trickle` and it does fail intermittently at fairly recent commit 8f1780f6dee2. Caught this one just now:
```
starting test - test_trickle_2_wait
Trickle 1 0x00400040 callback called

    Assertion failed at ZEPHYR_BASE/tests/net/trickle/src/main.c:171:
                 test_trickle_2_wait: (cb_2_called is false)

Trickle 2 no timeout
FAIL - test_trickle_2_wait
```
I don't know these tests. @andyross , @jukkar, anyone: does this assert look like a clock-related / `emu_time` issue or something else?

Reply:
> I don't know these tests. @andyross , @jukkar, anyone: does this assert look like a clock-related / `emu_time` issue or something else?

The test is clock related. While running the test in mps2_an385, I noticed that the test emits error print. I sent a #17289 that fixes the error printing but it does not touch the clock issues.
I could not reproduce the issue you mentioned after running the sanitychecker or doing "make run" quite many times in my host.

Reply:
I kept running this particular test locally and regularly and to be fair it seems to fail rarely. Now maybe it's just because the couple workarounds recently merged in sanitycheck made _all_ `emu_time` tests fail more rarely...

> I could not reproduce the issue you mentioned after running the sanitychecker or doing "make run" quite many times in my host.

Unless we're not talking about the same "make run", this runs only one test at a time AFAIK. For sanitycheck it depends on many factors, see @andyross analysis referenced above.